lib/git/lib.rb

```diff
@@ -764,11 +764,21 @@ def branch_delete(branch)
       command('branch', '-D', branch)
     end
 
+    # Runs checkout command to checkout or create branch
+    #
+    # accepts options:
+    #  :new_branch
+    #  :force
+    #  :start_point
+    #
+    # @param [String] branch
+    # @param [Hash] opts
     def checkout(branch, opts = {})
       arr_opts = []
       arr_opts << '-b' if opts[:new_branch] || opts[:b]
       arr_opts << '--force' if opts[:force] || opts[:f]
       arr_opts << branch
+      arr_opts << opts[:start_point] if opts[:start_point] && (arr_opts.include?('-b') || arr_opts.include?('--force'))
```

Comment:
In my reading of https://git-scm.com/docs/git-checkout, I think that either `-b`, `-B`, or `--orphan` is required to use `<start-point>`. I don't think that `--force` without `-b` would be a valid use case for adding `<start-point>`.

I think this line should be:

```ruby
  arr_opts << opts[:start_point] if opts[:start_point] && arr_opts.include?('-b')
```

Reply:
Hello James! Thank you for review.
You're correct, I was a bit mislead by `branch --force` https://git-scm.com/docs/git-branch#Documentation/git-branch.txt---force

This and other remarks should now be fixed in force-pushed branch.

Do not have a windows machine at hand for debugging, but the error looks like it should relate to the test running order and not windows itself, if so - probably should be fixed by test separation too.

Reply:
Looks great! Thank you.